core/src/main/java/org/apache/iceberg/hadoop/HadoopOutputFile.java

```diff
@@ -73,7 +87,7 @@ public Path getPath() {
   }
 
   public Configuration getConf() {
-    return conf;
+    return fs.getConf();
```

Comment:
This also needs to return the `Configuration` that was used to create the output file, if there was one.